R/generate_rules_matrix.R

```diff
@@ -6,70 +6,48 @@
 #'
 #' @param X The features matrix.
 #' @param rules_list A vector of causal rules.
-#' @param t The common support used in generating the causal rules matrix.
 #'
 #' @return
-#' a list with:
-#' - a raw matrix of causal rules
-#' - a standardized matrix of causal rules, and
-#' - a vector of causal rules
+#' the causal rules matrix.
 #'
-#' @keywords internal
 #'
-generate_rules_matrix <- function(X, rules_list, t) {
+generate_rules_matrix <- function(X, rules_list) {
 
-  # Generate and Rules Matrix
-  rules_matrix <- matrix(0, nrow = dim(X)[1], ncol = length(rules_list))
-  for (i in 1:length(rules_list)){
-    rules_matrix[eval(parse(text = rules_list[i])), i] <- 1
-  }
-
-  # Identify rules with too few observations
-  nrules <- dim(rules_matrix)[2]
-  ind <- 1:nrules
-
-  if (dim(X)[1] < 200){
-    t <- 0.05
-  }
-
-  sup <- apply(rules_matrix, 2, mean)
-  elim <- which((sup < t) | (sup > (1 - t)))
 
-  if (length(elim) > 0) {
-    ind <- ind[-elim]
+  # Generate and Rules Matrix
+  samplesize <- dim(X)[1]
+  nrules <- length(rules_list)
+  rules_matrix <- matrix(0, nrow = samplesize, ncol = nrules)
+  for (i in 1:nrules){
+    rules_matrix[eval(parse(text = rules_list[i]), list(X=X)), i] <- 1
   }
 
-  # Identify correlated rules
-  corelim <- 1
-  C <- stats::cor(rules_matrix[,ind])
-  diag(C) <- 0
-  nrules <- dim(rules_matrix[, ind])[2]
-  elim <- c()
-
-  for(i in 1:(nrules - 1)) {
-    elim <- c(elim, which(round(abs(C[i, (i + 1):nrules]),
-                                digits = 4) >= corelim) + i)
-  }
+  return(rules_matrix)
+}
 
-  if (length(elim) > 0) {
-    ind <- ind[-elim]
-  } else {
-    ind <- ind
-  }
 
-  # Remove rules with too few observations and correlated rules
-  rules_matrix <- rules_matrix[, ind,drop=FALSE]
-  rules_list <- rules_list[ind]
+#' @title
+#' Standardize Rules Matrix
+#'
+#' @description
+#' Standardize the matrix of causal rules given a list.
+#'
+#' @param rules_matrix The rules matrix.
+#'
+#' @return
+#' Standardized rules matrix
+#'
```

Comment:
If this is an internal function, please add `@keywords internal`.

Reply:
what's the difference between internal and external?

Reply:
We `@keywords internal` or `export`. Those functions that we do not want to share publicly (on CRAN) should be internal. Similar to the private method in other languages.

Reply:
No, actually this function should be public. If I not set anything is it public by default?

Reply:
just fixed